docs/guide/master/introduction/configuration.md
# Configuration

After installation, you can publish the configuration file using the following command:

```sh [artisan]
$ php artisan vendor:publish --provider="SequentSoft\ThreadFlow\Laravel\ServiceProvider"
```
This command will create a `thread-flow.php` configuration file in the `config` directory of your Laravel application.

::: details Example of the contents of this file
```php
'channels' => [
    'my-telegram-channel' => [
        'driver' => 'telegram',
        'session' => env('THREAD_FLOW_SESSION', 'database'),
        'dispatcher' => env('THREAD_FLOW_DISPATCHER', 'sync'),
        'entry' => \App\ThreadFlow\Pages\IndexPage::class,
        'api_token' => env('TELEGRAM_API_TOKEN', null),
        'webhook_secret' => env('TELEGRAM_WEBHOOK_SECRET', null),
        'timeout' => 30,
        'limit' => 100,
    ],
],

'sessions' => [
    'database' => [
        'driver' => 'eloquent',
        'model' => \SequentSoft\ThreadFlow\Laravel\Models\ThreadFlowSession::class,
    ],

    'cache' => [
        'driver' => 'cache',
        'store' => env('THREAD_FLOW_SESSION_CACHE_STORE', null),
        'max_lock_seconds' => 10,
        'max_lock_wait_seconds' => 15,
    ],

    'array' => [
        'driver' => 'array',
    ]
],


'dispatchers' => [
    'sync' => [
        'driver' => 'sync',
    ],
    'queue' => [
        'driver' => 'queue',
        'queue' => env('THREAD_FLOW_QUEUE', 'default'),
    ],
],
```
:::

## Channels

Contains a list of all channels that you want to use in your application.
Each channel has a unique key that you can use to reference it in your application.

You can use any key you want, and it does not have to match the name of the channel.
For example, you can use the `my-telegram-channel` key to reference the Telegram channel in your application.

## Driver

The `driver` option specifies the driver that should be used for the channel.
Yoy can use all installed drivers. For example, you can use the `telegram` driver for the Telegram channel.

::: info INFO
More information about drivers you can find [Drivers Page](/guide/master/drivers/). 
:::

## Session

The `session` option specifies the session store that should be used for the channel.
Available session stores: `database`, `cache`, `array` or any custom configured session store.

::: info INFO
More information about sessions you can find [Sessions Page](/guide/master/advanced/sessions).
:::


## Dispatcher

The `dispatcher` option specifies the dispatcher that should be used for the channel.
Available dispatchers: `sync` and `queue`.

::: info INFO
More information about dispatchers you can find [Dispatchers Page](/guide/master/advanced/dispatchers).
:::


## Entry

The `entry` option specifies the entry page that should be used for the channel.
This is the first page that will be displayed to the user when he starts the conversation.

## Other settings

Other settings are specific to the driver you are using.
For example, for the Telegram driver, you can use the `api_token`, `webhook_secret`, `timeout` and `limit` options.


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased]
- Adds first version

src/Laravel/config.php
<?php

return [
    /*
    |------------------------
    | Channels
    |------------------------
    |
    | Here you may define all the channels that ThreadFlow can use to communicate
    | with your users. ThreadFlow supports a variety of great notification services
    | which can all be used here. Supported drivers are: "telegram".
    |
    */
    'channels' => [

        'telegram' => [
            'driver' => 'telegram',
            'session' => env('THREAD_FLOW_SESSION', 'cache'),
            'active_pages' => 'cache',
            'pending_messages' => 'cache',
            'dispatcher' => 'sync',
            'entry' => \App\ThreadFlow\Pages\IndexPage::class,
            'api_token' => env('TELEGRAM_API_TOKEN'),
            'webhook_secret' => env('TELEGRAM_WEBHOOK_SECRET'),
            'timeout' => 30,
            'limit' => 100,
        ],

        'whatsapp' => [
            'driver' => 'whatsapp',
            'session' => env('THREAD_FLOW_SESSION', 'cache'),
            'active_pages' => 'cache',
            'pending_messages' => 'cache',
            'dispatcher' => 'sync',
            'entry' => \App\ThreadFlow\Pages\IndexPage::class,
            'from_phone_number_id' => env('WHATSAPP_FROM_PHONE_NUMBER_ID'),
            'api_token' => env('WHATSAPP_API_TOKEN'),
            'webhook_secret' => env('WHATSAPP_WEBHOOK_SECRET'),
            'timeout' => 30,
            'limit' => 100,
        ],
    ],

    /*
    |------------------------
    | Sessions
    |------------------------
    |
    | Here you may define all the session stores that ThreadFlow can use to store
    | user sessions. ThreadFlow supports a variety of great session stores
    | which can all be used here. Supported drivers are: "cache", "array" and "database".
    |
     */
    'sessions' => [
        'database' => [
            'driver' => 'eloquent',
            'model' => \SequentSoft\ThreadFlow\Laravel\Models\ThreadFlowSession::class,
        ],

        'cache' => [
            'driver' => 'cache',
            'store' => env('THREAD_FLOW_SESSION_CACHE_STORE', null),
            'max_lock_seconds' => 10,
            'max_lock_wait_seconds' => 15,
        ],

        'array' => [
            'driver' => 'array',
        ],
    ],

    /*
    |------------------------
    | Dispatchers
    |------------------------
    |
    | Here you may define all the dispatchers that ThreadFlow can use to handle
    | page dispatching. ThreadFlow supports a variety of great dispatchers
    | which can all be used here. Supported drivers are: "sync", "queue".
    |
     */
    'dispatchers' => [
        'sync' => [
            'driver' => 'sync',
        ],
        'queue' => [
            'driver' => 'queue',
            'queue' => env('THREAD_FLOW_QUEUE', 'default'),
        ],
    ],

    /*
     |------------------------
     | Active Pages
     |------------------------
     | Here you may define how to store pages that was visited but still active.
     | For example the pages you can go back to or pages for inline keyboards.
     | Supported drivers are: "cache", "array" and "eloquent".
     |
    */
    'active_pages' => [
        'database' => [
            'driver' => 'eloquent',
            'model' => \SequentSoft\ThreadFlow\Laravel\Models\ThreadFlowActivePage::class,
        ],
        'cache' => [
            'driver' => 'cache',
            'store' => env('THREAD_FLOW_ACTIVE_PAGES_CACHE_STORE', null),
        ],
        'array' => [
            'driver' => 'array',
        ],
    ],

    /*
     |------------------------
     | Pending Messages
     |------------------------
     | Here you may define how to store messages that was sent to user but user was busy at the moment.
     | For example the messages that was sent to user but user was filling a form.
     | Supported drivers are: "cache", "array" and "eloquent".
     |
    */
    'pending_messages' => [
        'database' => [
            'driver' => 'eloquent',
            'model' => \SequentSoft\ThreadFlow\Laravel\Models\ThreadFlowPendingMessage::class,
        ],
        'cache' => [
            'driver' => 'cache',
            'store' => env('THREAD_FLOW_PENDING_MESSAGES_CACHE_STORE', null),
        ],
        'array' => [
            'driver' => 'array',
        ],
    ],
];
